packages/pentaho/src/Canvas/SidePanel/useResizable.tsx

```diff
@@ -64,24 +64,33 @@ export const useResizable = (
   };
 
   const handleMouseUp = () => {
-    panelRef.current?.parentElement?.removeEventListener(
-      "mousemove",
-      mouseMove as unknown as EventListener,
-    );
-    panelRef.current?.parentElement?.removeEventListener(
-      "mouseup",
-      handleMouseUp as EventListener,
-    );
-    setIsDragging(false);
+    if (panelRef.current) {
+      panelRef.current.style.userSelect = "";
+      panelRef.current?.parentElement?.removeEventListener(
+        "mousemove",
+        mouseMove as unknown as EventListener,
+      );
+      panelRef.current?.parentElement?.removeEventListener(
+        "mouseup",
+        handleMouseUp as EventListener,
+      );
+      setIsDragging(false);
+    }
   };
 
   const startResizing = () => {
-    panelRef.current?.parentElement?.addEventListener(
-      "mousemove",
-      mouseMove as unknown as EventListener,
-    );
-    panelRef.current?.parentElement?.addEventListener("mouseup", handleMouseUp);
-    setIsDragging(true);
+    if (panelRef.current) {
+      panelRef.current.style.userSelect = "none";
+      panelRef.current.parentElement?.addEventListener(
+        "mousemove",
+        mouseMove as unknown as EventListener,
+      );
+      panelRef.current.parentElement?.addEventListener(
+        "mouseup",
+        handleMouseUp,
+      );
+      setIsDragging(true);
+    }
```

Comment:
Can we please use [guard clauses](https://deviq.com/design-patterns/guard-clause) so the code is easier to read (also because the lines are long)?

I typically wouldn't complain about this, but the types are also wrong! We should be using the global native `MouseEvent`, not `React.MouseEvent` (above too). With this change, there's no need for any `as` cast

```suggestion
    if (!panelRef.current) return;

    panelRef.current.style.userSelect = "none";
    panelRef.current.parentElement?.addEventListener("mousemove", mouseMove);
    panelRef.current.parentElement?.addEventListener("mouseup", handleMouseUp);
    setIsDragging(true);
```

Reply:
sure, updated the PR.